test/jellyfish/component/hls/request_handler_test.exs

```diff
@@ -46,15 +46,41 @@ defmodule Jellyfish.Component.HLS.RequestHandlerTest do
     assert {:ok, _pid} = RequestHandler.start(room_id)
   end
 
-  test "file request", %{room_id: room_id} do
-    add_mock_manifest(room_id)
+  test "file request - live stream", %{room_id: room_id} do
+    add_mock_manifest(room_id, persistent: false)
 
     assert {:ok, @manifest_content} == RequestHandler.handle_file_request(room_id, @manifest)
 
+    # Non persistent live streams and recordings resides in different folders
+    assert {:error, :recording_not_found} ==
+             RequestHandler.handle_recording_request(room_id, @manifest)
+
     assert {:error, :enoent} == RequestHandler.handle_file_request(room_id, @wrong_manifest)
-    assert {:error, :enoent} == RequestHandler.handle_file_request(@wrong_room_id, @manifest)
 
-    remove_mock_manifest(room_id)
+    assert {:error, :room_not_found} ==
+             RequestHandler.handle_file_request(@wrong_room_id, @manifest)
+
+    remove_mock_manifest(room_id, persistent: false)
+  end
+
+  test "file request - recordings", %{room_id: room_id} do
+    add_mock_manifest(room_id, persistent: true)
+
+    # Persistent live streams reside in the same directory as recordings.
+    # Nevertheless, the `RequestHandler` is capable of differentiating between them
+    assert {:error, :recording_not_found} ==
+             RequestHandler.handle_recording_request(room_id, @manifest)
+
+    assert {:ok, @manifest_content} == RequestHandler.handle_file_request(room_id, @manifest)
+
+    # When a room or HLS endpoint finishes, the corresponding path is removed from ETS.
+    # This implies that from this point the HLS stream becomes a recording.
+    EtsHelper.delete_hls_folder_path(room_id)
+
+    assert {:ok, @manifest_content} == RequestHandler.handle_recording_request(room_id, @manifest)
+    assert {:error, :room_not_found} == RequestHandler.handle_file_request(room_id, @manifest)
+
+    assert {:ok, @manifest_content}
   end
```

Comment:
Maybe we could move it into the `describe` block where we would define another setup function, where we would `add_mock_manifest` and `on_exit` remove it?